x/datadeal/keeper/deal.go

```diff
@@ -482,3 +482,30 @@ func (k Keeper) RemoveDataDeliveryVote(ctx sdk.Context, vote *types.DataDelivery
 
 	return nil
 }
+
+func (k Keeper) DeactivateDeal(ctx sdk.Context, msg *types.MsgDeactivateDeal) error {
```

Comment:
When deactivating deal, all the remained budget should be refunded to buyer as well.